src/main/java/audio/mixer/Mixer.java

```diff
@@ -0,0 +1,60 @@
+package audio.mixer;
+
+import java.util.ArrayList;
+
+public class Mixer {
+    /**
+     * Mixes the given audio streams together.
+     *
+     * @param audioStreams   list of double-arrays
+     * @param startPositions list of start positions for each audio stream (int[0] should always be 0)
+     * @return mixed audio stream as double-array
+     * @throws MixerException if the length of the audio streams does not match
+     */
+    public static double[] mixAudioStreams(ArrayList<double[]> audioStreams, int[] startPositions) throws MixerException {
+
+        // find the longest array
+        int maxLength = 0;
+        for (double[] audioStream : audioStreams) {
+            maxLength = Math.max(maxLength, audioStream.length);
+        }
+        // find the shortest array
+        int minLength = maxLength;
+        for (double[] audioStream : audioStreams) {
+            minLength = Math.min(minLength, audioStream.length);
+        }
```

Comment:
do we need to know the length of the shortest array? (and if yes, where?)

Reply:
Already removed that, see latest commit